app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    cached_results = {}
+
+    def wrapper(*args) -> Callable:
```

Comment:
The video tutorials talked about a tool that allows you not to lose the name of the decorated function and its docstring. This tool is in the functools library.